dataset.go

```diff
@@ -274,6 +279,44 @@ func (set dataset) Equal(other Data) bool {
 	return true
 }
 
+// see Data.Compare
+// Compare set and the otherSet, column by column, and calculate final result
+// for each row by merging columns comparison results.
+func (set dataset) Compare(other Data) ([]compare.Result, error) {
+	otherSet, ok := other.(Dataset)
+	if !ok {
+		return nil, ErrMismatchTypes
+	}
+	if set.Width() != otherSet.Width() {
+		return nil, errCompareDiffTypes
+	}
+
+	// Prepare first compare results, to be merged with future comparison results
+	res, err := set.At(0).Compare(otherSet.At(0))
+	if err != nil {
+		return nil, err
+	}
+	for i := 1; i < set.Width(); i++ {
+		iterationResult, err := set.At(i).Compare(otherSet.At(i))
+		if err != nil {
+			return nil, err
+		}
+		mergeComparisonResults(res, iterationResult)
+	}
+	return res, nil
+}
+
+func mergeComparisonResults(res, mergeWith []compare.Result) {
```

Comment:
can be understood from signature...
```suggestion
func merge(res, mergeWith []compare.Result) {
```